app/controllers/events_controller.rb

```diff
@@ -0,0 +1,75 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+  before_action :authorize_project_member
+  before_action :authorize_event_author, only: [:update, :destroy]
+
+  def index
+    @project = Project.find(params[:project_id])
+  end
+
+  def show
+  end
+
+  def new
+    @event = Project.find(params[:project_id]).events.build
+  end
+
+  def edit
+  end
+
+  def create
+    @event = Project.find(params[:project_id]).events.build(event_params)
+    @event.author = current_user
+    if @event.save
+      redirect_to(
+        [@event.project, @event],
+        notice: "Event was successfully created."
+      )
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @event.update(event_params)
+      redirect_to(
+        [@event.project, @event],
+        notice: "Event was successfully updated."
+      )
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @event.destroy
+    redirect_to(
+      project_events_path(@event.project),
+      notice: "Event was successfully destroyed."
+    )
+  end
+
+  private
+
+  def set_event
+    @event = Event.find_by_id(params[:id]) || raise_not_found
+  end
+
+  def event_params
+    params.require(:event).permit(:title, :description, :date)
+  end
+
+  def authorize_project_member
+    project_id = params[:project_id].to_i
+    raise_not_found unless Project.find_by_id(project_id)
+    unless current_user && current_user.takes_part_in_project?(project_id)
+      render status: :forbidden, plain: "403 Forbidden"
+    end
+  end
+
+  def authorize_event_author
+    unless current_user && current_user.events.find_by_id(params[:id])
```

Comment:
the difference is that `#find` raises error if no record is found, `#find_by_id` just returns `nil`